### PR TITLE
Add parent image to variants

### PR DESCRIPTION
**Pull request by pedromcp90**
Requested-by: [doofeeds/issues/544](https://github.com/doofinder/doofeeds/issues/544)
- Added the parent image if the variant image is empty.

### CODE REVIEW

File: doofinder-for-woocommerce/includes/class-rest-api-handler.php [hunk 1]
```diff
@@ -58,9 +58,17 @@ public static function add_woocommerce_product_images($result,  $server,  $reque
         ) {
             $products = $result->data;
             foreach ($products as $key => $product) {
+                if (!is_array($product))
+                    continue;
                 $post = get_post($product['id']);
                 $thumbnail = new Thumbnail($post);
                 $image_link = $thumbnail->get();
+                
+                //If the product is a variation and doesn't have image, take it form its parent
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/class-rest-api-handler.php:67]: There is a tiny mistake here: take it from its parent

File: doofinder-for-woocommerce/includes/class-rest-api-handler.php [hunk 2]
```diff
@@ -58,9 +58,17 @@ public static function add_woocommerce_product_images($result,  $server,  $reque
         ) {
             $products = $result->data;
             foreach ($products as $key => $product) {
+                if (!is_array($product))
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/class-rest-api-handler.php:61]: According to WP coding standards, curly braces are always required even for on-line statements: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#brace-style :smiley:

File: doofinder-for-woocommerce/includes/class-rest-api-handler.php [hunk 3]
```diff
@@ -58,9 +58,17 @@ public static function add_woocommerce_product_images($result,  $server,  $reque
...
+                    continue;
                 $post = get_post($product['id']);
                 $thumbnail = new Thumbnail($post);
                 $image_link = $thumbnail->get();
+                
+                //If the product is a variation and doesn't have image, take it form its parent
+                if (empty($image_link) && $post->post_type === 'product_variation') {
+                    $thumbnail = new Thumbnail(get_post($post->post_parent));
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/class-rest-api-handler.php:69]: Excellent approach!! :clap:

**Review by davidmolinacano — APPROVED** [review]
Great work!!! :medal_military:

**Review by eduardogomez97 — APPROVED** [review]

**Pull request merged**